File: packages/dubbo/src/__tests__/providers/com/alibaba/dubbo/demo/GenerticTypeResponse.ts
import java from 'js-to-java';

//generate by interpret-cli dubbo2.js

export interface IGenerticTypeResponse<T = any> {
  list?: Array<T>;
}

export class GenerticTypeResponse<T = any> {
  list?: Array<T>;

  constructor(params: IGenerticTypeResponse<T>) {
    this.list = params.list;
  }

  __fields2java() {
    return {
      $class: 'com.alibaba.dubbo.demo.GenerticTypeResponse',
      $: {
        list: this.list
          ? java.List(
              (this.list || []).map(paramItem => {
                return paramItem &&
                  (paramItem as {__fields2java?(): any})['__fields2java']
                  ? (paramItem as {__fields2java?(): any})['__fields2java']()
                  : paramItem;
              }),
            )
          : null,
      },
    };
  }
}

File: java/dubbo-demo/dubbo-demo-api/src/main/java/com/alibaba/dubbo/demo/GenerticTypeResponse.java
package com.alibaba.dubbo.demo;

import java.io.Serializable;
import java.util.List;

/**
 * Created by master on 21/11/2017.
 */
public class GenerticTypeResponse<T> implements Serializable {
    private List<T> list;

    public List<T> getList() {
        return list;
    }

    public void setList(List<T> list) {
        this.list = list;
    }

    @Override
    public String toString() {
        return "GenerticTypeResponse{" +
                "list=" + list +
                '}';
    }
}
